### ISSUE TITLE
Index Out Of Bound Error When Using MFXDatePicker

### ISSUE THREAD

**Issue by malintha-induwara** [palexdev/MaterialFX#352]
### Issue Description

A `IndexOutOfBoundsException` is thrown when using  MFXDatePicker

### Steps to Reproduce

1)Create a JavaFX application with an MFXDatePicker
2)Then just try to change the month or year while application is running

### MRE

```java

import io.github.palexdev.materialfx.controls.MFXDatePicker;
import io.github.palexdev.materialfx.utils.DateTimeUtils;
import io.github.palexdev.materialfx.utils.others.dates.DateStringConverter;
import javafx.fxml.FXML;
import javafx.fxml.Initializable;

import java.net.URL;
import java.util.ResourceBundle;

public class TestController implements Initializable {

    @FXML
    private MFXDatePicker dp;

    @Override
    public void initialize(URL location, ResourceBundle resources) {
        dp.setGridAlgorithm(DateTimeUtils::partialIntMonthMatrix);
        dp.setConverterSupplier(() -> new DateStringConverter("dd/MM/yyyy", dp.getLocale()));
    }
}

```
![error](https://github.com/palexdev/MaterialFX/assets/60071404/0b623329-a6fa-4f17-b8ad-d87eca5f34be)

**Comment by palexdev** [comment]
Is the DatePicker component correctly visualized? Is everything working?

**Comment by malintha-induwara** [comment]
Yes its correctly  visualized and everything is working fine except that exception.  It seems like the exception  isn't affecting anything it's just being thrown.

 _I created a simple JavaFX application and used DatePicker to record following demo. hope it helps you._

### Demo
https://github.com/palexdev/MaterialFX/assets/60071404/8c7275f5-002a-4839-9483-ec26bf5a5118

**Comment by aiwilson** [comment]
It does impact functionality. If you default the value of the date programmatically and click the icon, the calendar does not display the value of the defaulted date, instead it displays the current date. If you click the year dropdown, you will get this IndexOutOfBoundsException. If you select the year and month of the defaulted date, it will show the date is actually set. If you change the date using the UI, then it will jump to the correct date when the icon is clicked.

**Comment by palexdev** [comment]
Unfortunately, there's nothing I can do right now.
All controls using VirtualizedFX 11.x.x should be ported to the new 21.x.x version to work properly
